flink-runtime/src/main/java/org/apache/flink/runtime/io/network/NetworkEnvironment.java

```diff
@@ -228,7 +236,8 @@ public void releasePartitions(Collection<ResultPartitionID> partitionIds) {
 				resultPartitions[counter++] = resultPartitionFactory.create(taskName, executionId, rpdd);
 			}
 
-			registerOutputMetrics(outputGroup, buffersGroup, resultPartitions);
+			MetricGroup outputMetricGroup = taskMetricGroup.addGroup(METRIC_GROUP_OUTPUT);
+			registerOutputMetrics(outputMetricGroup, outputMetricGroup.addGroup(METRIC_GROUP_BUFFERS), resultPartitions);
```

Comment:
Maybe we could only pass `outputMetricGroup` here and make `outputMetricGroup.addGroup(METRIC_GROUP_BUFFERS)` inside `registerOutputMetrics`.